src/module-elasticsuite-core/Model/Autocomplete/Terms/DataProvider.php

```diff
@@ -97,17 +97,19 @@ public function getType()
     public function getItems()
     {
         if ($this->items === null) {
-            $collection = $this->getSuggestCollection();
             $this->items = [];
 
             if ($this->configurationHelper->isEnabled($this->getType())) {
-                foreach ($collection as $item) {
-                    $resultItem = $this->itemFactory->create([
-                        'title' => $item->getQueryText(),
-                        'num_results' => $item->getNumResults(),
-                        'type'        => $this->getType(),
-                    ]);
-                    $this->items[] = $resultItem;
+                $collection = $this->getSuggestCollection();
+                if ($collection !== null) {
```

Comment:
Perform the test `if ($this->queryFactory->get()->isQueryTextShort()) {` here and then initialize the collection.